test/unit/test_autosubmit_helper.py

```diff
@@ -17,81 +17,107 @@
 
 """Test file for ``autosubmit_helper.py``."""
 
-import datetime
-from datetime import timedelta
-from typing import Callable
+from datetime import datetime, timedelta
+from typing import Callable, Union
 
 import pytest
 
 import autosubmit.helpers.autosubmit_helper as helper
 from log.log import AutosubmitCritical
 
-
-@pytest.mark.parametrize('time', [
-    '04-00-00',
-    '04:00:00',
-    '2020:01:01 04:00:00',
-    '2020-01-01 04:00:00',
-    datetime.datetime.now() + timedelta(seconds=5),
-], ids=['wrong format hours', 'right format hours', 'fulldate wrong format', 'fulldate right format',
-        'execute in 5 seconds']
-                         )
-def teste_handle_start_time(time):
-    """
-    function to test the function handle_start_time inside autosubmit_helper
-    """
+_EXPID = 't000'
+
+
+@pytest.mark.parametrize(
+    'time',
+    [
+        '04-00-00',
+        '04:00:00',
+        '2020:01:01 04:00:00',
+        '2020-01-01 04:00:00',
+        datetime.now() + timedelta(seconds=5),
+    ],
+    ids=[
+        'wrong format hours',
+        'right format hours',
+        'fulldate wrong format',
+        'fulldate right format',
+        'execute in 5 seconds'
+    ]
+)
+def test_handle_start_time(time):
+    """Test the function handle_start_time inside autosubmit_helper."""
     if not isinstance(time, str):
         time = time.strftime("%Y-%m-%d %H:%M:%S")
     assert helper.handle_start_time(time) is None
 
 
-@pytest.mark.parametrize('ids, return_list_value, result', [
-    (None, [''], []),
-    ('', [''], ''),
-    ('a000', ['a001'], ''),
-    ('a000', ['a000'], ['a000']),
-    ('a000 a001', ['a000', 'a001'], ['a000', 'a001']),
-    ('a000 a001', ['a000', 'a001', 'a002'], ['a000', 'a001']),
-], ids=['None', 'expected AScritical members', 'expected AScritical rmembers',
-        'one ids', 'multiple sent ids', 'multiple return ids']
-                         )
-@pytest.mark.xfail(raise_stmt=AutosubmitCritical)
```

Comment:
The changes here, formatting & typos, were driven after I noticed this skipped test. In reality, the code is always going to raise exceptions when the test cases have conditions that do not pass the test for allowed members.

The test code was asserting for `''`, so I modified it to match what the code does, and instead assert that we have the `AutosubmitCritical`.